File: src/template/julia/fenwick/fenwick.jl
struct Fenwick
    n::Int
    bit::Vector{Int}
end

function Fenwick(n::Int)
    Fenwick(n, zeros(Int, n+1))
end

function update!(f::Fenwick, i::Int, v::Int)
    while i ≤ f.n
        f.bit[i] += v
        i += i & -i
    end
end

function query(f::Fenwick, i::Int)
    s = 0
    while i > 0
        s += f.bit[i]
        i -= i & -i
    end
    return s
end

range_query(f::Fenwick, l::Int, r::Int) = query(f, r) - query(f, l-1)

File: src/template/cpp/fenwick/fenwick.cpp
#include <vector>
using namespace std;

struct Fenwick {
    int n;
    vector<long long> bit;
    Fenwick(int _n): n(_n), bit(n+1, 0) {}
    void update(int i, long long delta) {
        for (; i <= n; i += i & -i) bit[i] += delta;
    }
    long long query(int i) const {
        long long s = 0;
        for (; i > 0; i -= i & -i) s += bit[i];
        return s;
    }
    long long rangeQuery(int l, int r) const {
        return query(r) - query(l - 1);
    }
};